File: binding/src/libqcolortransform.cpp
#include <QColor>
#include <QColorTransform>
#include <QRgba64>
#include <qcolortransform.h>
#include "libqcolortransform.h"
#include "libqcolortransform.hxx"

QColorTransform* QColorTransform_new() {
	 return new QColorTransform();
}

QColorTransform* QColorTransform_new2(const QColorTransform* colorTransform) {
	 return new QColorTransform(*colorTransform);
}

void QColorTransform_OperatorAssign(QColorTransform* self, const QColorTransform* other) {
	self->operator=(*other);
}

void QColorTransform_Swap(QColorTransform* self, QColorTransform* other) {
	self->swap(*other);
}

bool QColorTransform_IsIdentity(const QColorTransform* self) {
	return self->isIdentity();
}

unsigned int QColorTransform_Map(const QColorTransform* self, unsigned int argb) {
	return self->map(argb);
}

QRgba64* QColorTransform_Map2(const QColorTransform* self, QRgba64* rgba64) {
	return new QRgba64(self->map(*rgba64));
}

QRgbaFloat<qfloat16> QColorTransform_Map3(const QColorTransform* self, QRgbaFloat<qfloat16> rgbafp16) {
	return self->map(rgbafp16);
}

QRgbaFloat<float> QColorTransform_Map4(const QColorTransform* self, QRgbaFloat<float> rgbafp32) {
	return self->map(rgbafp32);
}

QColor* QColorTransform_Map5(const QColorTransform* self, const QColor* color) {
	return new QColor(self->map(*color));
}

void QColorTransform_Delete(QColorTransform* self) {
    delete self;
}

File: binding/include/libqcolortransform.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcolortransform.html
pub const qcolortransform = struct {

    /// New constructs a new QColorTransform object.
    pub fn New() QtC.QColorTransform {
        return qtc.QColorTransform_new();
    }


    /// New2 constructs a new QColorTransform object.
    pub fn New2(colorTransform: ?*anyopaque) QtC.QColorTransform {
        return qtc.QColorTransform_new2(@ptrCast(colorTransform));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QColorTransform_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QColorTransform_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn IsIdentity(self: ?*anyopaque, ) bool {
        return qtc.QColorTransform_IsIdentity(@ptrCast(self));
    }

    pub fn Map(self: ?*anyopaque, argb: u32) u32 {
        return qtc.QColorTransform_Map(@ptrCast(self), argb);
    }

    pub fn Map2(self: ?*anyopaque, rgba64: QtC.QRgba64) QtC.QRgba64 {
        return qtc.QColorTransform_Map2(@ptrCast(self), @ptrCast(rgba64));
    }

    pub fn Map5(self: ?*anyopaque, color: ?*anyopaque) QtC.QColor {
        return qtc.QColorTransform_Map5(@ptrCast(self), @ptrCast(color));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QColorTransform_Delete(@ptrCast(self));
    }
};
